spec/models/course_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+RSpec.describe Course, type: :model do
+it "has a valid factory" do
+  FactoryBot.build(:course).should be_valid
+end
+
+describe "enum" do
+  it { is_expected.to define_enum_for(:status).with_values(
+    pending: 0,
+    started: 1,
+    finished: 2
+  ) }
+end
+
+describe "validations" do
+  it { is_expected.to validate_presence_of(:name) }
+  it { is_expected.to validate_length_of(:name).is_at_most(50) }
+  it { is_expected.to validate_presence_of(:description)}
+  it { is_expected.to validate_presence_of(:status)}
+  it { is_expected.to validate_presence_of(:day_start)}
+  it { is_expected.to validate_presence_of(:day_end)}
+  it 'raises an error if end time is lower than start time' do
+    expect {FactoryBot.create(:course,day_start:DateTime.now+2.weeks,day_end:DateTime.now).to raise_error(ActiveRecord::Errors)}
+  end
+end
+
+describe "associations" do
+  it { is_expected.to have_many(:users) }
+  it { is_expected.to have_many(:course_users) }
+  it { is_expected.to have_many(:course_subjects) }
+  it { is_expected.to have_many(:subjects) }
+end
+describe "scopes" do
+ 
+  let(:course_one) {FactoryBot.create(:course,created_at: 2.weeks.ago)}
+  let(:course_two) {FactoryBot.create(:course,created_at: 1.weeks.ago)}
+  let(:course_search) {FactoryBot.create(:course,name:"buithikieu")}
+  let(:course_by_status) {FactoryBot.create(:course,status:1)}
+  it ".sort_by_created_at to sort the courses in descending order created_at" do
+   Course.sort_by_created_at.should eq [course_two, course_one]
+  end
+
+  it ".search_by_name to get course contain name" do
+    Course.search_by_name("bui").should include(course_search)
```

Comment:
Dùng include rồi lỡ cái scope k viết gì trong đó chẳng hạn, nó lấy hết toàn bộ record mà k lọc lại thì vẫn đúng?